### PR TITLE
Plugin Page: Content View

### PR DESCRIPTION
**Pull request by codemonkey800**
## Description

This PR implements the content view of the plugin page. This includes the page setup, setting up a plugin fixture for testing, and rendering the plugin name, summary, and description.

## Changes

- Adds JSON fixture for napari plugin [napari-compressed-labels-io](https://github.com/DragaDoncila/napari-compressed-labels-io).
- Markdown renderer using [react-markdown](https://github.com/remarkjs/react-markdown)
- Lazy loaded markdown syntax highlighting using [react-syntax-highlighter](https://github.com/react-syntax-highlighter/react-syntax-highlighter)
- Plugin page content view
- E2E test setup for plugin page

## Demos

**Demo Link**: https://napari-hub-content-view-demo.vercel.app/plugin/name

### Content View

image

### Syntax Highlighting

image

### Responsiveness

https://user-images.githubusercontent.com/2176050/115783610-ebf0f080-a371-11eb-912c-472a8870dcc8.mp4

## Follow Ups

- Connect plugin API to plugin page and replace fixture data with actual API data

### CODE REVIEW

**Comment by codemonkey800** [comment]
@kne42 Let me know if you would like for us to walk through the PR in case you don't understand a specific part of the React code. I'll try to find some time next week to go over React.

@tihuan I also added you as a reviewer, but your review is non-blocking for this PR 😄

File: client/src/components/PluginDetails/PluginDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  return <div />;
+}
+
+/**
+ * Component for rendering the plugin details page.
+ */
+export function PluginDetails(props: Props) {
+  return (
+    <div
+      data-testid="pluginDetails"
```
Comment (justinelarsen) [inline client/src/components/PluginDetails/PluginDetails.tsx:34]: Does "testid" need to be cleaned up?

**Review by justinelarsen — APPROVED** [review]

**Comment by goanpeca** [comment]
Random question unrelated to this PR, could we add something like https://www.netlify.com/ 
(or similar) to have live previews of the changes in a PR ?

File: client/src/components/common/Markdown/Markdown.module.scss [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    // Show scrollbar along x-axis for really long code
+    @apply overflow-x-auto;
+
+    // The Tailwind prose class adds a background around pre elements.
+    background: none;
+  }
+
+  img {
+    /*
+      Make images inline in markdown. By default, Tailwind will set all `img`
```
[the remark this answers is not in the document]
Comment (codemonkey800) [reply client/src/components/common/Markdown/Markdown.module.scss:12]: Yes that's correct! Some markdown like this:

```md
![](https://i.kym-cdn.com/entries/icons/original/000/013/564/doge.jpg)
```

will transform to this:

```html
<p>
  <img
    src="https://i.kym-cdn.com/entries/icons/original/000/013/564/doge.jpg"
    alt=""
  />
</p>
```

File: client/src/components/PluginDetails/PluginDetails.tsx (hunk 1 above)
Comment (codemonkey800) [reply client/src/components/PluginDetails/PluginDetails.tsx:34]: This is the established pattern by `react-testing-library` and unfortunately they don't allow us to change the key 😢 
https://testing-library.com/docs/queries/bytestid

**Comment by goanpeca** [comment]
>@goanpeca we've talked about using something like that with amplify, but we went with a different tech stack for deployment. something like vercel would probably be better since it has support for next.js apps and it's made by the same company.

Whatever allows us to see a live preview of changes in a PR will work :) , I just happen to know that one.

**Comment by codemonkey800** [comment]
@goanpeca I deployed a demo on Vercel for preview:
https://napari-hub-content-view-demo.vercel.app/plugin/name

**Comment by goanpeca** [comment]
Awesome @codemonkey800  :) 

Is there a way to automate this and get something that appears on the checks for a PR?

Screen Shot 2021-04-26 at 1 48 32 PM

This is what we have for example here: https://github.com/Quansight-Labs/quansight-labs-site/pull/184

Screen Shot 2021-04-26 at 1 49 35 PM

You get a direct link and everything is automated. It would really make the process of reviewing really nice :)

**Comment by justinelarsen** [comment]
This is awesome! Thanks @codemonkey800 and @goanpeca for improving the review process.

**Comment by codemonkey800** [comment]
@goanpeca There definitely is a way to [automate it using preview deployments](https://vercel.com/docs/platform/deployments#preview). It also looks like Netlify also has support for [next.js](https://www.netlify.com/with/nextjs/), but we currently don't have a contract with Vercel or Netlify for deployments. I spoke with @neuromusic about possibly using it, but I believe it would involve some additional process getting onboard with either.

We're currently using a different infrastructure for deploying the frontend though, so it would be interesting to see if it's something we could replicate. cc: @ziyangczi

**Pull request merged**